app/models/task.rb

```diff
@@ -29,10 +27,7 @@ def result
 
   def self.callback(msg)
     Task.transaction do
-      if job_id = fixer_callback_job_id(msg)
-        status = fixer_callback_status(msg)
-        time = fixer_callback_time(msg)
-      elsif job_id = porter_callback_job_id(msg)
+      if job_id = porter_callback_job_id(msg)
         status = porter_callback_status(msg)
```

Comment:
So a few lines below here, we only update the task if the status is non-nil.  And only JobReceived and JobResult get a status - so we're ignoring the TaskResults / TaskErrors.